docs/UserGuide.md

```diff
@@ -115,21 +115,58 @@ Examples:
 
 ### Locating persons by name: `find`
 
-Finds persons whose names contain any of the given keywords.
+Finds all contacts whose names or tags matches the substring keyword provided.
 
-Format: `find KEYWORD [MORE_KEYWORDS]`
 
-* The search is case-insensitive. e.g `hans` will match `Hans`
-* The order of the keywords does not matter. e.g. `Hans Bo` will match `Bo Hans`
-* Only the name is searched.
-* Only full words will be matched e.g. `Han` will not match `Hans`
-* Persons matching at least one keyword will be returned (i.e. `OR` search).
-  e.g. `Hans Bo` will return `Hans Gruber`, `Bo Yang`
+Format: `find FIELD/ KEYWORD FIELD/ KEYWORD ...`
+
+* 'FIELD' only supports `n/` for name and `t/` for tag.
+
+* 'KEYWORD' is the keyword to search for.
+
+* 'KEYWORD' can **ONLY** be alphabets (no spaces)
+  * `find n/John Doe` will **NOT** work. Try `find n/John n/Doe` instead to represent finding John and Doe
+
+* 'KEYWORD' and next 'FIELD' should be separated by a space.
+  * e.g. e.g `find n/John t/friends` will find all instances of John that have the tag friends 
+  * but `find n/Johnt/tfriends` will instead return an error since it assumes you are searching for.'Johnt/tfriends'
+  * and there should not be non-alphabet characters in the 'KEYWORD' field.
+
+* Multiple of the same 'FIELD's will be treated as a **logical AND**.
+  * e.g. `find n/John n/Doe` will return all instances of John and Doe.
+  * e.g `find n/Ale n/le` will still return instances ["Alex Liew", "Alexis Lebrun", "Alec"]
+
+* 'KEYWORD' should **NOT** be empty and there should be at least one 'FIELD' and 'KEYWORD' pair.
+  * e.g. `find n/ t/` and `find ` will **NOT** work.
+  
+* There shoud not be prefixes before the first 'FIELD' and 'KEYWORD' pair. 
+  * e.g. `find testing123 n/John` will **NOT** work.
+
+* Prefix Search: The search is a prefix search. 
```

Comment:
Just a nit - it seems that the find command is more of a substring search which is a more general form of prefix search since find n/ e returns ["Alex Yeoh", "Bernice Yu", "Charlotte Oliveiro"]